webclient/src/components/views/ProjectList.vue

```diff
@@ -14,31 +14,39 @@
         <Project v-for="project in projects" :project="project" :key="project.ID" v-on:update="Update" />
       </div>
     </div>
+    <div v-if="!loading">
+      <vue-simple-spinner></vue-simple-spinner>
+    </div>
   </div>
 </template>
 
 <script>
 import { GetProjects } from '@/database';
 import Project from '@/components/elements/Project';
 import LoggedInHeader from '@/components/elements/LoggedInHeader';
+import VueSimpleSpinner from 'vue-simple-spinner/src/components/Spinner';
 
 export default {
   components: {
+    VueSimpleSpinner,
     'Project': Project,
     'LoggedInHeader': LoggedInHeader
   },
   data () {
     return {
-      projects: []
+      projects: [],
+      loading: false
     };
   },
   methods: {
     Update: function () {
+      let self = this;
```

Comment:
You should be able to remove the `let self = this` and combine the `GetProjects()` to the `.then` that follows without storing it in a variable.